ptf/saiqosmap.py

```diff
@@ -0,0 +1,9596 @@
+                                 port,
+                                 qos_map_dict):
+        ''' Sets the multiple qos map ids attributes
+            for a given port.
+        Args:
+            port (obj_id): port object id
+            qos_map_dict (dict): qos map object ids dictionary
+        '''
+        params = [
+            'qos_pfc_priority_to_priority_group_map', 'qos_dot1p_to_tc_map',
+            'qos_tc_to_priority_group_map', 'qos_dscp_to_tc_map',
+            'qos_pfc_priority_to_queue_map'
+        ]
+        for key in qos_map_dict:
+            for p in params:
+                setattr(self, 't_%s' % p, None)
+            temp_name = 't_' + key
+            if key == 'qos_pfc_priority_to_priority_group_map':
+                setattr(self, temp_name, qos_map_dict[key])
+            if key == 'qos_dot1p_to_tc_map':
+                setattr(self, temp_name, qos_map_dict[key])
+            if key == 'qos_tc_to_priority_group_map':
+                setattr(self, temp_name, qos_map_dict[key])
+            if key == 'qos_dscp_to_tc_map':
+                setattr(self, temp_name, qos_map_dict[key])
+            if key == 'qos_pfc_priority_to_queue_map':
+                setattr(self, temp_name, qos_map_dict[key])
+            sai_thrift_set_port_attribute(
+                self.client,
+                port,
+                qos_tc_to_priority_group_map=getattr(
+                    self, 't_qos_tc_to_priority_group_map'),
+                qos_pfc_priority_to_priority_group_map=getattr(
+                    self, 't_qos_pfc_priority_to_priority_group_map'),
+                qos_pfc_priority_to_queue_map=getattr(
+                    self, 't_qos_pfc_priority_to_queue_map'),
+                qos_dot1p_to_tc_map=getattr(self, 't_qos_dot1p_to_tc_map'),
+                qos_dscp_to_tc_map=getattr(self, 't_qos_dscp_to_tc_map'))
+
+
+@group("draft")
+class L3QosDscpToTcTest(L3QosMapBaseClass):
+    """ L3 Test qos map Dscp to Tc Test attributes """
+    def setUp(self):
+        super(L3QosDscpToTcTest, self).setUp()
+
+    def runTest(self):
+        self.l3IPv4QosMapDscpToTcDefaultMappingTest()
+        self.l3IPv6QosMapDscpToTcDefaultMappingTest()
+        self.l3IPv4QosMapMultipleDscpToSingleTcMappingTest()
+        self.l3IPv6QosMapMultipleDscpToSingleTcMappingTest()
+        self.l3IPv4QosMapOneDscpToOneTcMappingTest()
+        self.l3IPv6QosMapOneDscpToOneTcMappingTest()
+        self.l3IPv4QosSameDscpToTcMappingManyPortsTest()
+        self.l3IPv6QosSameDscpToTcMappingManyPortsTest()
+        self.l3IPv4QosVariousDscpToTcMappingManyPortsTest()
+        self.l3IPv6QosVariousDscpToTcMappingManyPortsTest()
+
+    def tearDown(self):
+        super(L3QosDscpToTcTest, self).tearDown()
+
+    def l3IPv4QosVariousDscpToTcMappingManyPortsTest(self):
+        '''
+        L3 IPv4 DSCP to TC mapping for various ingress ports test.
+        '''
+        print("l3IPv4QosVariousDscpToTcMappingManyPortsTest")
```

Comment:
We are doing only routing traffic. Can you also add regular switching traffic for DSCP-->TC-->Queue.